parsing/dialects/dialect1/parserImpls.ftl

```diff
@@ -1792,6 +1792,55 @@ SqlNode SqlSelectTopN(SqlParserPos pos) :
     }
 }
 
+SqlNode InlineCaseSpecific() :
+{
+    final SqlNode value;
+    boolean not = false;
+}
+{
+    (
+        value = StringLiteral()
+    |
+        LOOKAHEAD( CompoundIdentifier() <LPAREN> [ <NOT> ] (<CASESPECIFIC>|<CS>) <RPAREN> )
+        value = CompoundIdentifier()
+    )
+    <LPAREN>
+    [ <NOT> { not = true; } ]
+    (
+        <CASESPECIFIC>
+    |
+        <CS>
+    )
+    <RPAREN>
+    {
+        return new SqlInlineCaseSpecific(getPos(), not, value);
+    }
+}
+
+/* This has to be separate from the InlineCaseSpecific() due to the LOOKAHEAD
+   for preExpressionMethods in Parser.jj breaking if both CompoundIdentifier()
+   and NamedFunctionCall() are options.
+ */
+SqlNode InlineCaseSpecificNamedFunctionCall() :
+{
+    final SqlNode value;
+    boolean not = false;
+}
+{
+    value = NamedFunctionCall()
+    <LPAREN>
```

Comment:
From line 1831 down to 1841 can all go into the helper function mentioned above (e.g. called "CaseSpecific").